openstack/datadog_checks/openstack/openstack.py

```diff
@@ -1202,68 +1201,60 @@ def check(self, instance):
-            #  instances
-            projects = []
             for _, scope in scope_map.iteritems():
                 # Store the scope on the object so we don't have to keep passing it around
                 self._current_scope = scope
 
                 self._send_api_service_checks(scope, custom_tags)
 
-                collect_all_projects = is_affirmative(instance.get("collect_all_projects", False))
-                collect_all_tenants = is_affirmative(instance.get('collect_all_tenants', False))
-
                 self.log.debug("Running check with credentials: \n")
                 self.log.debug("Nova Url: %s", self.get_nova_endpoint())
                 self.log.debug("Neutron Url: %s", self.get_neutron_endpoint())
 
-                # Restrict monitoring to this (host, hypervisor, project)
-                # and it's guest servers
-
-                hyp = self.get_local_hypervisor()
-
                 project = self.get_scoped_project(scope)
+                projects[project['name']] = project
 
-                if collect_all_projects or project is None:
-                    scope_projects = self.get_all_projects(scope)
-                    if scope_projects:
-                        projects.extend(scope_projects)
-                else:
-                    projects.append(project)
+                # self.get_stats_for_single_project(project, custom_tags)
 
-                # Restrict monitoring to non-excluded servers
                 i_key = self._instance_key(instance)
-                servers = self.get_servers_managed_by_hypervisor(
-                    i_key, collect_all_tenants, split_hostname_on_first_period=split_hostname_on_first_period,
-                )
 
-                host_tags = self._get_tags_for_host(split_hostname_on_first_period=split_hostname_on_first_period)
+            if collect_limits_from_all_projects:
+                scope_projects = self.get_all_projects(scope)
+                if scope_projects:
+                    for proj in scope_projects:
+                        projects[proj['name']] = proj
 
-                # Deep copy the cache so we can remove things from the Original during the iteration
-                server_cache_copy = copy.deepcopy(self.server_details_by_id)
+            proj_filtered = pattern_filter(
+                                           [p for p in projects],
+                                           whitelist=self.include_project_name_rules,
+                                           blacklist=self.exclude_project_name_rules
+                                           )
+            proj_copy = copy.deepcopy(projects)
+            for proj in proj_copy:
+                if proj not in proj_filtered:
+                    del projects[proj]
 
-                for server in server_cache_copy:
-                    server_tags = copy.deepcopy(custom_tags)
-                    server_tags.append("nova_managed_server")
+            for proj in projects:
```

Comment:
can we do `for name, project in projects.iteritems` and call the function below passing just `project`?